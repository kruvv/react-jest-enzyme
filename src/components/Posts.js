import React from 'react';
import Post from './Post';

const NEWS = [
    {
        author: "Viktor",
        created_at: "2020-03-01T13:10:45",
        num_comments: 10,
        title: "Enzyme",
        objectId: 1,
        points: 10,
        url: "//test.url",
    },
    {
        author: "Ivan",
        created_at: "2019-09-21T10:20:05",
        num_comments: 8,
        title: "Jest",
        objectId: 2,
        points: 110,
        url: "//test2121.url",
    },
    {
        author: "Vadim",
        created_at: "2012-05-22T23:40:05",
        num_comments: 4,
        title: "JavaScript",
        objectId: 5,
        points: 20,
        url: "//main.urls",
    },
];

const Posts = () =>  (
        <ul className="newsList">
            {NEWS.map(({author, created_at, num_comments, title, points, url, objectId}) => (
                <Post
                    key={objectId}
                    author={author}
                    created_at={created_at}
                    num_comments={num_comments}
                    title={title}
                    points={points}
                    url={url}
                />
            ))}
        </ul>
    );


export default Posts;
