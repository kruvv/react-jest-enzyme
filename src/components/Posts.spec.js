import { shallow } from 'enzyme';
import React from 'react';
import Posts from './Posts';
import { cleanup } from '@testing-library/react';

const setUp = (props) => shallow(<Posts {...props} />)
describe('<Posts />', () => {
    let component;
    beforeEach(() => {
        component = setUp();
    })

    afterEach(cleanup);
    it("should correctly render component Posts", () => {
        expect(component).toMatchSnapshot();
        // console.log(component.debug());
    })
})
