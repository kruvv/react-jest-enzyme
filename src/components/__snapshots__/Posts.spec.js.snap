// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should correctly render component Posts 1`] = `
<ul
  className="newsList"
>
  <Post
    author="Viktor"
    created_at="2020-03-01T13:10:45"
    key="1"
    num_comments={10}
    points={10}
    title="Enzyme"
    url="//test.url"
  />
  <Post
    author="Ivan"
    created_at="2019-09-21T10:20:05"
    key="2"
    num_comments={8}
    points={110}
    title="Jest"
    url="//test2121.url"
  />
  <Post
    author="Vadim"
    created_at="2012-05-22T23:40:05"
    key="5"
    num_comments={4}
    points={20}
    title="JavaScript"
    url="//main.urls"
  />
</ul>
`;
